lib/screens/product_edit_screen.dart

```diff
@@ -63,167 +70,369 @@ class _TBProductEditScreenState extends State<TBProductEditScreen> {
 
   @override
   Widget build(BuildContext context) {
+    AsyncValue<TBProductModel> selectedProduct =
+        ref.watch(productProvider(widget.id));
```

Comment:
Same use final and rename it to productState. There is no need to watch it, you just read the data.